src/form/Input.js

```diff
@@ -0,0 +1,159 @@
+                this.checkNodeName('SELECT',element);
+                this._inputElement.attr('value', this.value());
+                this.insertSelectOptions(this.selectOptions());
+                break;
+            case 'textarea':
+                this.checkNodeName('TEXTAREA',element);
+                this._inputElement.node().innerHTML = this.value();
+                break;
+            case 'button':
+                this.checkNodeName('BUTTON',element);
+                this._inputElement.node().innerHTML = this.value();
+                break;
+            case 'checkbox':
+                this.checkNodeName('INPUT',element);
+                this._inputElement.property('checked', this.value());
+                break;
+            default:
+                this.checkNodeName('INPUT',element);
+                this._inputElement.attr('value', this.value());
+                this._inputElement.attr('type', this.type());
+                break;
+        }
+        if(typeof(this._blur) !== 'function'){
+            this._blur = function(){};
+        }
+        if(typeof(this._click) !== 'function'){
+            this._click = function(){};
+        }
+        if(typeof(this._change) !== 'function'){
+            this._change = function(){};
+        }
+        this._inputElement.on('change',function(w){
+            w.resetValue(w);
+            w._change();
+        });
+        this._inputElement.on('click',function(w){
+            d3.event.preventDefault();
+            w._click(w);
+        });
+        this._inputElement.on('blur',function(w){
+            w._blur(w);
+        });
+    };
+    
+    Input.prototype.insertSelectOptions = function (optionsArr) {
+        var optionHTML = '';
+        if(optionsArr.length > 0){
+            optionsArr.forEach(function(opt){
+                optionHTML += '<option value="'+opt+'">'+opt+'</option>';
+            });
+        } else {
+            optionHTML += '<option>selectOptions not set</option>';
+        }
+        this._inputElement.node().innerHTML = optionHTML;
+    };
+    
+    Input.prototype.checkNodeName = function (expected,elm) {
+        var node = this._inputElement.node();
+        if(node.nodeName !== expected){
+            node.remove();
+            this.enter(null,elm);
+        }
+    };
+    
+    Input.prototype.resetValue = function (w) {
+        if(w.type() === "checkbox"){
+            w.value(w._inputElement.node().checked);
+        } else {
+            w.value(w._inputElement.node().value);
+        }
+    };
+    
+    Input.prototype.isValid = function(){
+        if(typeof (this.validate()) === 'string'){
+            var re = new RegExp(this.validate());
+            if (!re.test(this.value())) {
+                return false;
+            }
+        }
+        return true;
+    };
+    
+    Input.prototype.blur = function (_) {
+        if (!arguments.length){
+            this._blur();
+        } else {
+            this._blur = _;
```

Comment:
I am not sure what your trying to achieve with the "_blur"?  In general this "Event" function would be just an empty stub, which is called when the input event files - it is up to the user of this widget to wire it up?